mlkem/poly.h

```diff
@@ -324,7 +324,36 @@ ENSURES(ARRAY_IN_BOUNDS(int, k, 0, MLKEM_N - 1, r->coeffs, 0, MLKEM_Q - 1));
 // clang-format on
 
 #define poly_add MLKEM_NAMESPACE(poly_add)
-void poly_add(poly *r, const poly *a, const poly *b);
+/************************************************************
+ * Name: poly_add
+ *
+ * Description: Computes the coefficient-wise addition of two polynomials
+ *
+ * Arguments: - r: Pointer to output polynomial
+ *            - a: Pointer to first input polynomial. Must be the same as r.
+ *            - b: Pointer to second input polynomial. Must be disjoint from a
+ *                 and r.
+ *
+ * The coefficients of a and b must be so that the addition does
+ * not overflow. Otherwise, the behaviour of this function is undefined.
+ *
+ * NOTE: The restriction r == a is not needed, but we currently only
+ *       prove a CBMC spec under that assumption.
+ *
+ ************************************************************/
+void poly_add(poly *r, const poly *a, const poly *b)
+    // clang-format off
+// So far, poly_add is only used with r == a, so restrict proof to this case
+// TODO: Is this the right way to capture the aliasing constraint a==r?
```

Comment:
Can @rod-chapman comment on this, please?

Reply:
I pinged the CBMC team about this and would file a follow-up PR if the current formulation is not idiomatic. Ok?

Reply:
https://github.com/pq-code-package/mlkem-c-aarch64/issues/323